otherlibs/unix/unix.mli

```diff
@@ -675,11 +675,39 @@ val close_process_full :
 (** {6 Symbolic links} *)
 
 
-val symlink : string -> string -> unit
-(** [symlink source dest] creates the file [dest] as a symbolic link
-   to the file [source].
-
-   On Windows: not implemented. *)
+val symlink : ?is_dir:bool -> string -> string -> unit
+(** [symlink ?is_dir source dest] creates the file [dest] as a symbolic link
+   to the file [source]. On Windows, [~is_dir] indicates if the symbolic link
+   points to a directory or a file; if omitted, [symlink] examines [source]
+   using [stat] and picks appropriately, if [source] does not exist then [false]
+   is assumed. On Unix, [~is_dir] ignored.
```

Comment:
I don't think this is a good behaviour. `is_dir` should always be ignored whenever the target exists, it should only be used whenever the target doesn't exist. Note that alternatively we could eschew the parameter and disallow symlinking to non existent targets on Windows.